tests/rptest/tests/consumer_group_recovery_tool_test.py

```diff
@@ -41,10 +43,36 @@ def __init__(self, test_ctx, *args, **kwargs):
             node_prealloc_count=1,
             **kwargs)
 
-    def describe_all_groups(self):
+    def describe_all_groups(self, num_groups: int = 1):
         rpk = RpkTool(self.redpanda)
         all_groups = {}
-        for g in rpk.group_list():
+
+        # The one consumer group in this test comes from KgoVerifierConsumerGroupConsumer
+        # for example, kgo-verifier-1691097745-347-0
+        kgo_group_re = re.compile(r'^kgo-verifier-[0-9]+-[0-9]+-0$')
+
+        self.logger.debug(f"Issue ListGroups, expect {num_groups} groups")
+
+        def do_list_groups():
+            res = rpk.group_list()
+
+            if res is None:
+                return False
+
+            if len(res) != num_groups:
+                return False
+
+            kgo_group_m = kgo_group_re.match(res[0])
+            self.logger.debug(f"kgo group match {kgo_group_m}")
+            return False if kgo_group_m is None else (True, res)
```

Comment:
did you intend to check that all the groups had the expected name format?

i mean, i think the check as-is is strong enough for this PR, just wondering.

Reply:
> did you intend to check that all the groups had the expected name format?
> 
> i mean, i think the check as-is is strong enough for this PR, just wondering.

Yes I did intend to check that.
This test should create a single group coming from the `KgoVerifierConsumerGroupConsumer`. So I wanted to check the scenarios where:
(A) a consumer group that is not from kgo-verifier (That's the point of the regex match)
(B) 2 or more consumer groups from the kgo-verifier (that's the point of the `len(res)` check)

So now the test will check for (A) and (B)

Reply:
cool 👍